pkg/apis/core/validation/validation.go

```diff
@@ -2503,6 +2503,72 @@ func validatePullPolicy(policy core.PullPolicy, fldPath *field.Path) field.Error
 	return allErrors
 }
 
+func validateEphemeralContainers(ephemeralContainers []core.EphemeralContainer, containers, initContainers []core.Container, volumes map[string]core.VolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(ephemeralContainers) == 0 {
+		return allErrs
+	}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.EphemeralContainers) {
+		return append(allErrs, field.Forbidden(fldPath, "disabled by EphemeralContainers feature-gate"))
+	}
+
+	allNames := sets.String{}
+	for _, c := range containers {
+		allNames.Insert(c.Name)
+	}
+	for _, c := range initContainers {
+		allNames.Insert(c.Name)
+	}
+
+	for i, ec := range ephemeralContainers {
+		idxPath := fldPath.Index(i)
+		specPath := idxPath.Child("spec")
+
+		if ec.TargetContainerName != "" && !allNames.Has(ec.TargetContainerName) {
+			allErrs = append(allErrs, field.NotFound(idxPath.Child("targetContainerName"), ec.TargetContainerName))
+		}
+
+		if ec.Spec.Name == "" {
+			allErrs = append(allErrs, field.Required(specPath, "ephemeralContainer requires a name"))
+			continue
+		}
+
+		// Using validateContainers() here isn't ideal because it adds an index to the error message that
+		// doesn't really exist for EphemeralContainers (i.e. ephemeralContainers[0].spec[0].name instead
+		// of ephemeralContainers[0].spec.name)
+		// TODO(verb): factor a validateContainer() out of validateContainers() to be used here
+		allErrs = append(allErrs, validateContainers([]core.Container{ec.Spec}, false, volumes, specPath)...)
+		// EphemeralContainers don't require the backwards-compatibility distinction between pod/podTemplate validation
+		allErrs = append(allErrs, validateContainersOnlyForPod([]core.Container{ec.Spec}, specPath)...)
+
+		if allNames.Has(ec.Spec.Name) {
+			allErrs = append(allErrs, field.Duplicate(specPath.Child("name"), ec.Spec.Name))
+		} else {
+			allNames.Insert(ec.Spec.Name)
+		}
+
+		if ec.Spec.Lifecycle != nil {
```

Comment:
forbidding specific fields like this means we are likely to accidentally allow new fields added to v1.Container without considering whether it makes sense for ephemeral containers. at the very least, we need a reflective test that makes us consider whether to prevent new fields from being set on ephemeral containers as they are added, or we could switch to a whitelist approach in what we allow here